### DISCUSSION TITLE
Using Serial on my own server

### DISCUSSION THREAD

**Discussion by andreachat** [BIPES/BIPES#117] [General]
Hello ! Thanks for your great work !

I installed BIPES on my own server. I'm trying to use the serial connexion instead of wifi. 
The serial button redirect on the official bipes website (https://bipes.net.br/ide/).

I tried to change urls on the ui/core/channel.js file (urls on this.ifunavailable attribut) with my own server adress. So now it redirects to my server but with the wifi connexion again.

Is it possible to use Serial on my own server ? And if yes how ?

Thanks for reading :)

**Comment by gastmaier** [comment]
Hi Andréa!

Your fix in *ui/core/channel.js* is absolutely correct.
I think the issue is that you haven't set up the server to allow HTTPS, you must set up a certificate to access HTTPS even in a local network.

Note: With localhost (http://127.0.0.1/beta/2): you should be able to connect with Wi-Fi and serial under HTTP.

You can learn [how to set up a self-signed certificate with Apache 2 here](https://www.digitalocean.com/community/tutorials/how-to-create-a-self-signed-ssl-certificate-for-apache-in-ubuntu-16-04).
It's important to note that your browser **will** complain about the self-signed certificate, but you can ignore or [add "yourself" as a trusted certificate authority to your browser](https://www.techrepublic.com/article/how-to-add-a-trusted-certificate-authority-certificate-to-chrome-and-firefox/).

**Reply by andreachat** [reply]
Thanks for your fast reply !

Everything is clear now, I will try to setup my HTTPS certificate, I should have done it anyway :smile: 
Thanks also for the very useful links :+1:

**Reply by andreachat** [reply]
It works perfectly :ok_hand:
